controllers/propagator/policy_controller.go

```diff
@@ -122,20 +123,63 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return reconcile.Result{}, err
 			}
 
+			if len(replicatedPlcList.Items) == 0 {
+				log.V(2).Info("No replicated policies to delete.")
+
+				return reconcile.Result{}, nil
+			}
+
+			policiesChan := make(chan policiesv1.Policy, len(replicatedPlcList.Items))
+			deletionResultsChan := make(chan deletionResult, len(replicatedPlcList.Items))
+
+			var numWorkers int
```

Comment:
Optional: Since this section is essentially copying code from the handle decision concurrency, it might be nice to have a small helper function called `getNumWorkers` which accepts a length of a slice.